pkg/cloud/cloud.go

```diff
@@ -1,37 +1,429 @@
+	output, err := c.fsx.CreateFileCacheWithContext(ctx, input)
 	if err != nil {
-		return nil, fmt.Errorf("could not get metadata from AWS: %v", err)
+		if isIncompatibleParameter(err) {
+			return nil, ErrFcExistsDiffSize
+		}
+		return nil, fmt.Errorf("CreateFileCache failed: %v", err)
 	}
 
-	return &cloud{
-		metadata: metadata,
+	if output.FileCache.LustreConfiguration.MountName == nil {
+		return nil, fmt.Errorf("mountName returned in CreateFileCache is nil")
+	}
+	mountName := *output.FileCache.LustreConfiguration.MountName
+
+	perUnitStorageThroughput := int64(0)
+	if output.FileCache.LustreConfiguration.PerUnitStorageThroughput != nil {
+		perUnitStorageThroughput = *output.FileCache.LustreConfiguration.PerUnitStorageThroughput
+	}
+
+	return &FileCache{
+		FileCacheId:              *output.FileCache.FileCacheId,
+		CapacityGiB:              *output.FileCache.StorageCapacity,
+		DnsName:                  *output.FileCache.DNSName,
+		MountName:                mountName,
+		FileCacheType:            *output.FileCache.FileCacheType,
+		FileCacheTypeVersion:     *output.FileCache.FileCacheTypeVersion,
+		perUnitStorageThroughput: perUnitStorageThroughput,
 	}, nil
 }
 
-func (c *cloud) GetMetadata() MetadataService {
-	return c.metadata
+func (c *cloud) DeleteFileCache(ctx context.Context, fileCacheId string) (err error) {
+	input := &fsx.DeleteFileCacheInput{
+		FileCacheId: aws.String(fileCacheId),
+	}
+	if _, err = c.fsx.DeleteFileCacheWithContext(ctx, input); err != nil {
+		if isFileCacheNotFound(err) {
+			return ErrNotFound
+		}
+		return fmt.Errorf("DeleteFileCache failed: %v", err)
+	}
+	return nil
+}
+
+func (c *cloud) DescribeFileCache(ctx context.Context, fileCacheId string) (*FileCache, error) {
+	fc, err := c.getFileCache(ctx, fileCacheId)
+	if err != nil {
+		return nil, err
+	}
+
+	if fc.LustreConfiguration.MountName == nil {
+		return nil, fmt.Errorf("mountName returned in DescribeFileCache is nil")
+	}
+	mountName := *fc.LustreConfiguration.MountName
+
+	perUnitStorageThroughput := int64(0)
+	if fc.LustreConfiguration.PerUnitStorageThroughput != nil {
+		perUnitStorageThroughput = *fc.LustreConfiguration.PerUnitStorageThroughput
+	}
+
+	return &FileCache{
+		FileCacheId:              *fc.FileCacheId,
+		CapacityGiB:              *fc.StorageCapacity,
+		DnsName:                  *fc.DNSName,
+		MountName:                mountName,
+		FileCacheType:            *fc.FileCacheType,
+		FileCacheTypeVersion:     *fc.FileCacheTypeVersion,
+		perUnitStorageThroughput: perUnitStorageThroughput,
+	}, nil
+}
+
+func (c *cloud) WaitForFileCacheAvailable(ctx context.Context, fileCacheId string) error {
+	err := wait.Poll(PollCheckInterval, PollCheckTimeout, func() (done bool, err error) {
+		fc, err := c.getFileCache(ctx, fileCacheId)
+		if err != nil {
+			return true, err
+		}
+		klog.V(2).Infof("WaitForFileCacheAvailable filesystem %q status is: %q", fileCacheId, *fc.Lifecycle)
```

Comment:
Super nit - do we want to log with verbosity set to 2 here? Seems a bit excessive